libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.h

```diff
@@ -0,0 +1,157 @@
+ @param window The window that was brought to the front
+ */
+- (void)windowManager:(SFSDKWindowManager*)windowManager didBringToFront:(SFSDKWindowContainer *)window;
+
+/**
+ Called when the controller will be presented.
+ @param windowManager The window manager making this call
+ @param window The window that will be used to present a controller
+ @param controller The controller that will be presented
+ */
+- (void)windowManager:(SFSDKWindowManager*)windowManager willPushViewController:(SFSDKWindowContainer *)window controller:(UIViewController *)controller;
+
+/**
+ Called when the controller is presented.
+ @param windowManager The window manager making this call
+ @param window The window that was used to present a controller
+ @param controller The controller that was presented
+ */
+- (void)windowManager:(SFSDKWindowManager*)windowManager didPushViewController:(SFSDKWindowContainer *)window controller:(UIViewController *)controller;
+
+/**
+ Called when the controller will be be dismissed.
+ @param windowManager The window manager making this call
+ @param window The window that will be used to dismiss a controller
+ @param controller The controller that was presented
+ */
+- (void)windowManager:(SFSDKWindowManager*)windowManager willPopViewController:(SFSDKWindowContainer *)window controller:(UIViewController *)controller;
+
+/**
+ Called when the controller is dismissed
+ @param windowManager The window manager making this call
+ @param window The window that was used to dismiss a controller
+ @param controller The controller that was dismissed
+ */
+- (void)windowManager:(SFSDKWindowManager*)windowManager didPopViewController:(SFSDKWindowContainer *)window controller:(UIViewController *)controller;
+@end
+
+@interface SFSDKWindowManager : NSObject
+
+/** SDK uses this window to present the login flow.
+ */
+@property(readonly,strong) SFSDKWindowContainer *authWindow;
+
+/** SDK uses this window to present the snapshot View.
+ */
+@property(readonly,nonatomic,strong) SFSDKWindowContainer *snapshotWindow;
+
+/** SDK uses this window to present the passocde View.
+ */
+@property(readonly,nonatomic,strong) SFSDKWindowContainer *passcodeWindow;
+
+/** Use this to customize the snapshotview
+ */
+@property(nonatomic,strong) UIViewController *snapshotViewController;
+
+@property (nonatomic, strong,readonly) NSMapTable<NSString *,SFSDKWindowContainer *> *namedWindows;
+
+/** Api to push viewcontroller into a given window. BringtoFront the window & then push
+ */
+- (void)pushViewController:(UIViewController *)controller window:(SFSDKWindowContainer *)window withCompletion:(void (^)(void))completion;
+
+/** Api to pop viewcontroller from a given window.
+ */
+- (void)popViewController:(UIViewController *)controller window:(SFSDKWindowContainer *)window withCompletion:(void (^)(void))completion;
+
+/** Returns the SFSDKWindowContainer window representing the mainWindow that has been set
+ */
+- (SFSDKWindowContainer *)mainWindow;
```

Comment:
Why are this and `setMainWindow:` defined explicitly, rather than exposing a property?